pkg/apis/serving/v1alpha2/predictor.go

```diff
@@ -144,6 +158,42 @@ func setResourceRequirementDefaults(requirements *v1.ResourceRequirements) {
 	}
 }
 
+func validateResourceRequirement(requirements *v1.ResourceRequirements) error {
```

Comment:
We will need this validation for predictor/explainer/transformer. Is there a way we can generalize?

Reply:
I see and will add this for explainer/transformer. Now the only thing needed is the func to transfer the type of requirements for the validation function to use.
```func transferToCoreResourceRequirements(rr *v1.ResourceRequirements) *core.ResourceRequirements``` I'm thinking to add this a common place so that explainer/transformer can call this function too.

Reply:
I meant to put the validation logic to a common place but I saw there's validateStorageURI function which is used by explainer too. I think it's fine since it's in same package so will follow.